File: platform/polyglot/extract-imports-text-based-notation/java/src/main/java/com/internal/extractimports/Logger.java
package com.internal.extractimports;

import java.util.logging.ConsoleHandler;
import java.util.logging.Formatter;
import java.util.logging.Level;
import java.util.logging.LogRecord;

/**
 * Package-local logger with prefixed formatting.
 */
public final class Logger {

    private final java.util.logging.Logger delegate;

    private Logger(java.util.logging.Logger delegate) {
        this.delegate = delegate;
    }

    /**
     * Create a logger with package name and filename prefix.
     */
    public static Logger create(String packageName, String filename) {
        String name = packageName + ":" + filename;
        java.util.logging.Logger jLogger = java.util.logging.Logger.getLogger(name);

        if (jLogger.getHandlers().length == 0) {
            ConsoleHandler handler = new ConsoleHandler();
            String prefix = "[" + packageName + ":" + filename + "]";
            handler.setFormatter(new Formatter() {
                @Override
                public String format(LogRecord record) {
                    return prefix + " " + record.getMessage() + System.lineSeparator();
                }
            });
            jLogger.addHandler(handler);
            jLogger.setUseParentHandlers(false);
        }

        jLogger.setLevel(Level.WARNING);
        return new Logger(jLogger);
    }

    public void debug(String msg) {
        delegate.fine(msg);
    }

    public void info(String msg) {
        delegate.info(msg);
    }

    public void warn(String msg) {
        delegate.warning(msg);
    }

    public void error(String msg) {
        delegate.severe(msg);
    }
}

File: platform/polyglot/extract-imports-text-based-notation/mjs/src/logger.ts
export interface LoggerInstance {
  debug(msg: string, ...args: unknown[]): void;
  info(msg: string, ...args: unknown[]): void;
  warn(msg: string, ...args: unknown[]): void;
  error(msg: string, ...args: unknown[]): void;
}

export interface LoggerOptions {
  writer?: {
    debug: (...args: unknown[]) => void;
    info: (...args: unknown[]) => void;
    warn: (...args: unknown[]) => void;
    error: (...args: unknown[]) => void;
  };
}

export function create(
  packageName: string,
  filename: string,
  options?: LoggerOptions,
): LoggerInstance {
  const prefix = `[${packageName}:${filename}]`;
  const writer = options?.writer ?? console;

  return {
    debug: (msg: string, ...args: unknown[]) =>
      writer.debug(prefix, msg, ...args),
    info: (msg: string, ...args: unknown[]) =>
      writer.info(prefix, msg, ...args),
    warn: (msg: string, ...args: unknown[]) =>
      writer.warn(prefix, msg, ...args),
    error: (msg: string, ...args: unknown[]) =>
      writer.error(prefix, msg, ...args),
  };
}
